### ISSUE TITLE
SoGo Mail shows wrong timecode

### ISSUE THREAD

**Issue by McMeanF** [mailcow/mailcow-dockerized#2520]
I have updated every possible TZ in probably every location I can find, but no matter what I do my email timestamp in SoGo is always one hour behind.

I have set TZ=Europe/London in mailcow.conf

When I issue the date command in Ubuntu it gives me the correct time and timezone and syncing is enabled and working.

The calendar in the web ui gives the correct time, but emails, no matter what are always an hour behind.

I have looked in syslogs and they are showing the correct time, but I think the postfix logs are also an hour behind....

Thanks

**Comment by andryyy** [comment]
You just change it in mailcow.conf. You should never ever change it in other locations. I cannot help you as I don't know what was changed now...

You should have checked the config file "sogo.conf" or the SOGo container (does it show the correct time stamps?). Now that you changed ALL files with "TZ", it is difficult.

**Comment by McMeanF** [comment]
Yes chap

I set it in the mailcow.conf and have checked all containers. All are reporting correct time, but still in the web ui the sent and received mail is an hour behind... 

The calendar shows the correct time, just sent and received mail is wrong..

**Comment by McMeanF** [comment]
Just to give you a bit more info:

****************************  sogo.conf ****************************

{
    SOGoCalendarDefaultRoles = (
        PublicViewer,
        ConfidentialDAndTViewer,
        PrivateDAndTViewer
    );

    WOWorkersCount = "20";
    SOGoACLsSendEMailNotifications = YES;
    SOGoAppointmentSendEMailNotifications = YES;
    SOGoDraftsFolderName = "Drafts";
    SOGoJunkFolderName= "Junk";
    SOGoMailDomain = "sogo.local";
    SOGoEnableEMailAlarms = YES;
    SOGoFoldersSendEMailNotifications = YES;
    SOGoForwardEnabled = YES;
    SOGoUIAdditionalJSFiles = (js/custom-sogo.js);
    SOGoEnablePublicAccess = YES;

    // Multi-domain setup
    // Domains are isolated, you can define visibility options here.
    // Example:

    // SOGoDomainsVisibility = (
    //  (domain1.tld, domain5.tld),
    //  (domain3.tld, domain2.tld)
    // );

    SOGoSieveServer = "sieve://dovecot:4190/?tls=YES";
    SOGoSMTPServer = "postfix:588";
    WOPort = "0.0.0.0:20000";
    SOGoMemcachedHost = "memcached";

    SOGoLanguage = English;
    SOGoMailAuxiliaryUserAccountsEnabled = YES;
    SOGoMailCustomFromEnabled = YES;
    SOGoMailingMechanism = smtp;
    SOGoSMTPAuthenticationType = plain;

    SxVMemLimit = 384;

    SOGoMaximumPingInterval = 3540;

    SOGoInternalSyncInterval = 45;
    SOGoMaximumSyncInterval = 3540;

    // 100 seems to break some Android clients
    //SOGoMaximumSyncWindowSize = 99;
    // This should do the trick for Outlook 2016
    SOGoMaximumSyncResponseSize = 512;

    WOWatchDogRequestTimeout = 20;
    WOListenQueueSize = 300;
    WONoDetach = YES;

    SOGoIMAPAclConformsToIMAPExt = Yes;
    SOGoPageTitle = "SOGo Groupware";
    SOGoFirstDayOfWeek = "1";

    SOGoSieveFolderEncoding = "UTF-8";
    SOGoPasswordChangeEnabled = YES;
    SOGoSentFolderName = "Sent";
    SOGoMailShowSubscribedFoldersOnly = NO;
    NGImap4ConnectionStringSeparator = "/";
    SOGoSieveScriptsEnabled = YES;
    SOGoTrashFolderName = "Trash";
    SOGoVacationEnabled = YES;

    SOGoCacheCleanupInterval = 900;
    SOGoMaximumFailedLoginCount = 10;
    SOGoMaximumFailedLoginInterval = 900;
    SOGoFailedLoginBlockInterval = 900;

    MySQL4Encoding = "utf8mb4";
  //SOGoDebugRequests = YES;
  //SoDebugBaseURL = YES;
  //ImapDebugEnabled = YES;
  //SOGoEASDebugEnabled = YES;
  //LDAPDebugEnabled = YES;
  //PGDebugEnabled = YES;
  //MySQL4DebugEnabled = YES;
  //SOGoUIxDebugEnabled = YES;
  //WODontZipResponse = YES;
    WOLogFile = "/dev/sogo_log";
}


********************************* mailcow.conf **********************************

# ------------------------------
# mailcow web ui configuration
# ------------------------------
# example.org is _not_ a valid hostname, use a fqdn here.
# Default admin user is "admin"
# Default password is "moohoo"

MAILCOW_HOSTNAME=hermes.mcmean.net

# ------------------------------
# SQL database configuration
# ------------------------------

DBNAME=mailcow
DBUSER=mailcow

# Please use long, random alphanumeric strings (A-Za-z0-9)

DBPASS=VKgRzqnDxaaB7jANTw5tjFo409LL
DBROOT=c8zvTcKIr6pjIcKZdPyHNAqpY7Y1

# ------------------------------
# HTTP/S Bindings
# ------------------------------

# You should use HTTPS, but in case of SSL offloaded reverse proxies:

HTTP_PORT=480
HTTP_BIND=0.0.0.0

HTTPS_PORT=4443
HTTPS_BIND=0.0.0.0

# ------------------------------
# Other bindings
# ------------------------------
# You should leave that alone
# Format: 11.22.33.44:25 or 0.0.0.0:465 etc.
# Do _not_ use IP:PORT in HTTP(S)_BIND or HTTP(S)_PORT

SMTP_PORT=25
SMTPS_PORT=465
SUBMISSION_PORT=587
IMAP_PORT=143
IMAPS_PORT=993
POP_PORT=110
POPS_PORT=995
SIEVE_PORT=4190
DOVEADM_PORT=127.0.0.1:19991
SQL_PORT=127.0.0.1:13306

# Your timezone

TZ=Europe/London

# Fixed project name

COMPOSE_PROJECT_NAME=mailcowdockerized

# Set this to "allow" to enable the anyone pseudo user. Disabled by default.
# When enabled, ACL can be created, that apply to "All authenticated users"
# This should probably only be activated on mail hosts, that are used exclusivly by one organisation.
# Otherwise a user might share data with too many other users.
ACL_ANYONE=disallow

# Garbage collector cleanup
# Deleted domains and mailboxes are moved to /var/vmail/_garbage/timestamp_sanitizedstring
# How long should objects remain in the garbage until they are being deleted? (value in minutes)
# Check interval is hourly

MAILDIR_GC_TIME=1440

# Additional SAN for the certificate
#
# You can use wildcard records to create specific names for every domain you add to mailcow.
# Example: Add domains "example.com" and "example.net" to mailcow, change ADDITIONAL_SAN to a value like:
#ADDITIONAL_SAN=imap.*,smtp.*
# This will expand the certificate to "imap.example.com", "smtp.example.com", "imap.example.net", "imap.example.net"
# plus every domain you add in the future.
#
# You can also just add static names...
#ADDITIONAL_SAN=srv1.example.net
# ...or combine wildcard and static names:
#ADDITIONAL_SAN=imap.*,srv1.example.com
#

ADDITIONAL_SAN=

# Skip running ACME (acme-mailcow, Let's Encrypt certs) - y/n

SKIP_LETS_ENCRYPT=y

# Skip IPv4 check in ACME container - y/n

SKIP_IP_CHECK=n

# Skip HTTP verification in ACME container - y/n

SKIP_HTTP_VERIFICATION=n

# Skip ClamAV (clamd-mailcow) anti-virus (Rspamd will auto-detect a missing ClamAV container) - y/n

SKIP_CLAMD=y

# Skip Solr on low-memory systems or if you do not want to store a readable index of your mails in solr-vol-1.
SKIP_SOLR=y

# Solr heap size in MB, there is no recommendation, please see Solr docs.
# Solr is a prone to run OOM and should be monitored. Unmonitored Solr setups are not recommended.
SOLR_HEAP=1024

# Enable watchdog (watchdog-mailcow) to restart unhealthy containers (experimental)

USE_WATCHDOG=n

# Allow admins to log into SOGo as email user (without any password)

ALLOW_ADMIN_EMAIL_LOGIN=n

# Send notifications by mail (no DKIM signature, sent from watchdog@MAILCOW_HOSTNAME)
# Can by multiple rcpts, NO quotation marks

#WATCHDOG_NOTIFY_EMAIL=a@example.com,b@example.com,c@example.com
#WATCHDOG_NOTIFY_EMAIL=

# Max log lines per service to keep in Redis logs

LOG_LINES=9999

# Internal IPv4 /24 subnet, format n.n.n (expands to n.n.n.0/24)

IPV4_NETWORK=172.22.1

# Internal IPv6 subnet in fc00::/7

IPV6_NETWORK=fd4d:6169:6c63:6f77::/64

# Use this IPv4 for outgoing connections (SNAT)

#SNAT_TO_SOURCE=

# Use this IPv6 for outgoing connections (SNAT)

#SNAT6_TO_SOURCE=

# Create or override API key for web uI
# You _must_ define API_ALLOW_FROM, which is a comma separated list of IPs
# API_KEY allowed chars: a-z, A-Z, 0-9, -

#API_KEY=
#API_ALLOW_FROM=127.0.0.1,1.2.3.4

# mail_home is ~/Maildir
MAILDIR_SUB=Maildir

I can, if need be supply main.cf and master.cf from the sogo container and docker-compose.yml if you need.


I appreciate the help

**Comment by andryyy** [comment]
It only applies to mail?

Can you open a ticket with SOGo?

------ Originalnachricht ------
Von: "McMeanF" <notifications@github.com>
An: "mailcow/mailcow-dockerized" <mailcow-dockerized@noreply.github.com>
Cc: "André Peters" <andre.peters@debinux.de>; "Comment" 
<comment@noreply.github.com>
Gesendet: 12.04.2019 10:24:45
Betreff: Re: [mailcow/mailcow-dockerized] SoGo Mail shows wrong timecode 
(#2520)

**Comment by McMeanF** [comment]
That's correct. Only the emails show the incorrect time when they are sent or received.

I will raise a ticket with SOGo as suggested

Do you need anything else from me?

**Comment by andryyy** [comment]
A `docker-compose down` and `docker-compose up -d` does not change that behaviour?

The timezone within postfix-mailcow and sogo-mailcow is correct? (See `docker-compose logs --tail=200 postfix-mailcow sogo-mailcow`

**Comment by McMeanF** [comment]
Yea, ran the <code>docker-compose up</code> and <code>docker-compose down -d</code> commands, no joy.

I ran the following and this is what was returned. Shows the correct time

<code>sudo docker exec -it "Sogo-ID" date</code>
<code>Fri Apr 12 09:37:23 BST 2019</code>

<code>sudo docker exec -it "Postfix-ID" date</code>
<code>Fri Apr 12 09:41:11 BST 2019</code>

<code>sudo docker exec -it "Sogo-ID" cat /etc/timezone</code>
<code>Etc/UTC</code>

<code>sudo docker exec -it "Postfix-ID" cat /etc/timezone</code>
<code>Etc/UTC</code>


NB

I have replaced the numerical container IDs with something easier to read

**Comment by McMeanF** [comment]
I have used the <code>docker run -e TZ=Europe/London mailcow/sogo:1.54 date</code> command and it returns the correct date, but when I do down/up and restart the containers, it then displays the incorrect date..


Maybe if I use a Dockerfile setting to override this at the start of the container?

**Comment by McMeanF** [comment]
Added this to the end of the SOGo and Postfix Dockerfiles

<code>RUN sudo echo "Europe/London" > /etc/timezone</code>
<code>RUN sudo dpkg-reconfigure -f noninteractive tzdata</code>

I rebooted and used <code>sudo docker -it "Container-ID" date</code> and now it returning the wrong time code....

**Comment by andryyy** [comment]
A reboot does nothing. You would need to rebuild the image and recreate the container. I cannot help when you change the containers, I don' know what is all messed up now. Just start over clean (remove ALL containers and images), set the TZ vars and try again. We need the logs from unchanged containers and images.

It reads the TZ var from the mailcow.conf file. So setting it in mailcow.conf is enough. It might be a symlink issue with mailcow.conf to .env. Perhaps a permission issue?

Do you start the stack as root?

**Comment by McMeanF** [comment]
You're just a few minutes too late.....

I've just done exactly what you suggested with starting from scratch.

Purged all the containers and started again.

I made sure that my host system TZ was correct and then when it asks me for the TZ when generating the conf file I entered my TZ and low and behold all is well with the universe I believe. 

Just need to test the emails and see what dates show, but I think it's going to be okay....


Lesson learnt, enter you TZ at installation and don't change anything.

Thanks for the help

**Comment by badsmoke** [comment]
Hello, I have the same problem.

The time zone is not correct.

mailcow.conf:


```
# Your timezone

TZ=Europe/Berlin
```

```
docker exec -it mailcowdockerized_postfix-mailcow_1 cat /etc/timezone:
Etc/UTC
```

```
docker exec -it mailcowdockerized_sogo-mailcow_1 cat /etc/timezone:
Etc/UTC
```

Time is set correctly on both 
 
Host computer also has the time zone Europe/Berlin

For appointment invitations Etc/UTC is unfortunately displayed.

docker-compose down &&& docker-compose up -d was executed

**Comment by andryyy** [comment]
docker-compose exec sogo-mailcow /bin/bash
echo $TZ

**Comment by badsmoke** [comment]
```
docker-compose exec sogo-mailcow /bin/bash
WARNING: The WATCHDOG_NOTIFY_EMAIL variable is not set. Defaulting to a blank string.
root@0072454ea08c:/# echo $TZ
Europe/Berlin
```

**Comment by andryyy** [comment]
Please also run this:

```
docker-compose exec sogo-mailcow /bin/bash
grep -i time /var/lib/sogo/GNUstep/Defaults/sogod.plist -n2
```

What's the output here?

**Comment by badsmoke** [comment]
```
17-    <key>SOGoProfileURL</key>
18-    <string>mysql://mailcow:4BgovHiD9gR8GMKrQUl6fEl3ImKo@%2Fvar%2Frun%2Fmysqld%2Fmysqld.sock/mailcow/sogo_user_profile</string>
19:    <key>SOGoTimeZone</key>
20-    <string>Europe/Berlin</string>
21-    <key>domains</key>
```

**Comment by andryyy** [comment]
How do you create the appointment invitations? Which client?

**Comment by badsmoke** [comment]
With Outlook (CalDav Plugin)
Standard Mail App Windows 10 
Thunderbird

**Comment by andryyy** [comment]
Created in SOGo:
![image](https://user-images.githubusercontent.com/2972950/57059942-cb788900-6cb7-11e9-8ff5-66d676ee7580.png)

Received on GMail:
![image](https://user-images.githubusercontent.com/2972950/57059962-d7fce180-6cb7-11e9-89ce-a21941944788.png)

Created in emClient:
![image](https://user-images.githubusercontent.com/2972950/57060005-0084db80-6cb8-11e9-855f-b92538c8ba43.png)

Received on GMail:
![image](https://user-images.githubusercontent.com/2972950/57060026-185c5f80-6cb8-11e9-8485-c9f4bdae4b25.png)

**Comment by andryyy** [comment]
Does it work when you create them in SOGo?

**Comment by badsmoke** [comment]
Ok, directly via SOGo it works as intended with Europe/Berlin

![sogo](https://user-images.githubusercontent.com/24853712/57062081-6a54b380-6cbf-11e9-8b00-8d05e2f39982.png)


but with the other clients: Etc/UTC

![outlook](https://user-images.githubusercontent.com/24853712/57062071-658fff80-6cbf-11e9-919a-53f94c687ea8.png)

**Comment by andryyy** [comment]
Not here.

Report it to SOGo then. I won’t debug it.

**Comment by badsmoke** [comment]
from the SOGo , it works, is it really a SOGo problem?

**Comment by andryyy** [comment]
Not sure. I’d say the clients set the time zone. If it is missing, SOGo perhaps adds this field.

I cannot debug this for you. Try other extensions or check the plugins for time zone settings. :) emClient sends them fine, too.

**Comment by andryyy** [comment]
Is Thunderbird set to auto select the time zone? Does your Windows use a different time zone?

**Comment by andryyy** [comment]
Please also try this. But don’t use ActiveSync in this plugin. The EAS version is not supported. Just use DAV:

https://github.com/jobisoft/TbSync/wiki/How-to-get-started

**Comment by pnt3x** [comment]
Hey guys,
I'm facing the same error.
After updating my system with the update script, the e-mails I receive have the wrong time. 
I have updated from version 2024-01e to version 2024-04.
The time zone in mailcow.conf has remained unchanged (`TZ=Europe/Berlin`).

When I display the time zone in the containers, it is set to Etc/UTC:
`docker exec -it container /bin/bash --> cat /etc/timezone`

I have integrated the system time as a volume to all containers:
`/etc/timezone:/etc/timezone:ro`
`/etc/localtime:/etc/localtime:ro`
The emails are still delivered with the wrong time included.

The correct time is displayed in SoGO. In Outlook, Thunderbird, ... not.
